d2l-input-time.html

```diff
@@ -0,0 +1,392 @@
+		<span class="timezone">{{timezone}}</span>
+	</template>
+
+	<script>
+	/* global d2lIntl */
+	(function() {
+		var instances = 0;
+		Polymer({
+
+			is: 'd2l-input-time',
+
+			behaviors: [
+				Polymer.IronA11yKeysBehavior
+			],
+
+			properties: {
+				timeInterval: {
+					type: Number,
+					value: 30
+				},
+				hours: {
+					type: Number,
+					notify: true,
+					value: 23
+				},
+				minutes: {
+					type: Number,
+					notify: true,
+					value: 59
+				},
+				value: {
+					type: String,
+					notify: true,
+					observer: '_valueChanged'
+				},
+				locale: {
+					type: String,
+					value: ''
+				},
+				overrides: {
+					type: Object,
+					value: {}
+				},
+				timezone: String,
+				formatter: {
+					type: Object,
+					readOnly: true,
+					value: new d2lIntl.DateTimeFormat()
+				},
+				parser: {
+					type: Object,
+					readOnly: true,
+					value: new d2lIntl.DateTimeParse()
+				},
+				times: {
+					type: Object,
+					readOnly: true,
+					value: []
+				},
+				boundary: Object,
+				_dropdownWidth: String,
+				listboxId: String,
+				selectedListboxId: String,
+				label: String
+			},
+
+			observers: [
+				'_updateFormater(locale, overrides)',
+				'updateValue(hours, minutes)',
+				'updateSelection(hours, minutes)',
+				'_updateTimes(timeInterval, formatter)'
+			],
+
+			listeners: {
+				'iron-select': '_scrollSelectedItemIntoView'
+			},
+
+			keyBindings: {
+				'esc': '_onTimeInputKeyPressed',
+				'enter': '_onTimeInputKeyPressed',
+				'up': '_onTimeInputKeyPressed',
+				'down': '_onTimeInputKeyPressed',
+				'home': '_onTimeInputKeyPressed',
+				'end': '_onTimeInputKeyPressed'
+			},
+
+			clear: function() {
+				this.updateValue(23, 59);
+			},
+
+			// time display logic start
+
+			_updateFormater: function(locale, overrides) {
+				this._setFormatter(new d2lIntl.DateTimeFormat(locale, { locale: overrides }));
+				this._setParser(new d2lIntl.DateTimeParse(locale, { locale: overrides }));
+				this.updateValue(this.hours, this.minutes);
+			},
+
+			_valueChanged: function(value) {
+				this._closeTimesList(false);
+				var date = this.parser.parseTime(value);
+				if (date) {
+					this._dontUpdateValue = true;
+					this.hours = date.getHours();
+					this.minutes = date.getMinutes();
+					this.fire('d2l-input-time-changed', {
+						hours: this.hours,
+						minutes: this.minutes,
+						value: this.value
+					});
+					this._dontUpdateValue = false;
+				}
+			},
+
+			// Update the value given the specified hours and minutes
+			updateValue: function(hours, minutes) {
+				hours = +hours;
+				minutes = +minutes;
+				if (!this.formatter || this._dontUpdateValue) {
+					return;
+				}
+				if (hours >= 0 && minutes >= 0) {
+					var dateDisplay = new Date(0, 0, 0, hours || 0, minutes || 0, 0, 0);
+					this.value = this.formatter.formatTime(dateDisplay);
+				}
+			},
+
+			attached: function() {
+				document.body.addEventListener('focus', this.__onAutoCloseFocus, true);
```

Comment:
Do we know what's different about the focus/close logic here vs what dropdown provides?  It would be really great if this component did not need to provide custom logic for this, since it's typically the source of many bugs, especially when considering shadow-DOM vs shady.

Reply:
Removing this causes the menu not to disappear when someone tabs off the input without selecting anything, or clicks elsewhere on the page.  It may be related to the opener not being on the right level?

Added https://github.com/BrightspaceUI/inputs/issues/23

Reply:
Yeah, it could be because the focusable opener element is not the one marked at the opener.  Not sure.  Sounds like these scenarios should be handled by dropdown though.